cdap-watchdog/src/main/java/co/cask/cdap/logging/write/LogCleanup.java

```diff
@@ -86,33 +90,49 @@ Location getParent(Location location) {
   }
 
   /**
-   * Deletes dir if it is empty, and recursively deletes parent dirs if they are empty too. The recursion stops at
-   * non-empty parent or base directory. If dir is not child of base directory then the recursion stops at root.
-   * @param dir dir to be deleted.
+   * For the specified directory to be deleted, finds its namespaced log location, then deletes
+   * @param dir
+   * @throws IOException
    */
-  void deleteEmptyDir(Location dir) {
+  void deleteEmptyDir(Location dir) throws IOException {
     LOG.debug("Got path {}", dir.toURI());
 
+    List<Location> namespaceDirs = rootDir.list();
```

Comment:
Agree its a little dirty, but the problem is that ``LogCleanup`` does not know about the namespace. Let me think. Maybe we could pass it in.